source/agora/common/BlockStorage.d

```diff
@@ -761,6 +788,80 @@ public class BlockStorage : IBlockStorage
         {
         }
     }
+
+    /*******************************************************************************
+
+        Read the bytes to calculate the checksum
+
+        Params:
+            data = file data bytes
+
+        Returns:
+            calculate the checksum
+
+    *******************************************************************************/
+
+    private ubyte[] checksum (const ubyte[] data) @safe nothrow
+    {
+        auto crc32 = new CRC32Digest();
```

Comment:
`auto` => `scope` to allocate on the stack instead of the heap.

Reply:
Fixed